plotly/figure_factory/_facet_grid.py

```diff
@@ -783,7 +783,7 @@ def create_facet_grid(df, x, y, facet_row=None, facet_col=None,
                     )
     show_legend = False
     if color_name:
-        if isinstance(df[color_name][0], str) or color_is_cat:
+        if isinstance(df[color_name].iloc[0], str) or color_is_cat:
```

Comment:
@alishobeiri There's a `elif isinstance(df[color_name][0], Number):` line a little further down the code that you can change too.

Thanks for doing this! Awesome stuff.